container_opts.go

```diff
@@ -127,3 +127,16 @@ func setSnapshotterIfEmpty(c *containers.Container) {
 		c.Snapshotter = DefaultSnapshotter
 	}
 }
+
+// WithContainerExtension appends extension data to the container object.
+// Use this to decorate the container object with additional data.
+func WithContainerExtension(extension interface{}) NewContainerOpts {
+	return func(ctx context.Context, client *Client, c *containers.Container) error {
+		any, err := typeurl.MarshalAny(extension)
```

Comment:
This will require the urls to be registered with `typeurl.Register` unless they are proto messages. May want to specify so in the godoc

Reply:
Done.